modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -0,0 +1,707 @@
+    private final static String userDelegate = "will";
+    private int port;
+    private String hostname = "";
+    private String serviceURL = "";
+
+
+    public ActivitiRestClient(String portM, String hostnameM) {
+        port = Integer.parseInt(portM);
+        hostname = hostnameM;
+        serviceURL = "http://" + hostnameM + ":" + portM + "/bpmnrest/";
+    }
+
+    /**
+     * Used put the thread in sleep until the tasks are generated
+     */
+    public void waitForTaskGeneration() throws InterruptedException {
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException ie) {
+            log.error(" Thread Interuppted", ie);
+        }
+    }
+
+
+    /**
+     * This Method is used to deploy BPMN packages to the BPMN Server
+     *
+     * @param fileName The name of the Package to be deployed
+     * @param filePath The location of the BPMN package to be deployed
+     * @throws java.io.IOException
+     * @throws org.json.JSONException
+     * @returns String array with status, deploymentID and Name
+     */
+    public String[] deployBPMNPackage(String filePath, String fileName)
+            throws IOException, JSONException {
+        String url = serviceURL + "repository/deployments";
+
+        HttpHost target = new HttpHost(hostname, port, "http");
+
+        DefaultHttpClient httpClient = new DefaultHttpClient();
+        httpClient.getCredentialsProvider().setCredentials(
+                new AuthScope(target.getHostName(), target.getPort()),
+                new UsernamePasswordCredentials(username, password)
+        );
+        try {
+
+            HttpPost httpPost = new HttpPost(url);
+
+            MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+            builder.addBinaryBody("file", new File(filePath),
+                                  ContentType.MULTIPART_FORM_DATA, fileName);
+            HttpEntity multipart = builder.build();
+            httpPost.setEntity(multipart);
+            HttpResponse response = httpClient.execute(httpPost);
+            String status = response.getStatusLine().toString();
+            String responseData = EntityUtils.toString(response.getEntity());
+            JSONObject jsonResponseObject = new JSONObject(responseData);
+            if (status.contains(Integer.toString(HttpStatus.SC_CREATED)) || status.contains(Integer.toString(HttpStatus.SC_OK))) {
+                String deploymentID = jsonResponseObject.getString("id");
+                String name = jsonResponseObject.getString("name");
+                return new String[]{status, deploymentID, name};
+            }
+            if (status.contains(Integer.toString(HttpStatus.SC_INTERNAL_SERVER_ERROR))) {
+                String errorMessage = jsonResponseObject.getString("errorMessage");
+                return new String[]{status, errorMessage};
+            }
+        } catch (IOException ioMessage) {
+            log.error("Data Post Failure", ioMessage);
```

Comment:
As all these exceptions are thrown from this method (i.e. declared in the method signature), don't catch them here. If it is not possible to handle an exception or add more context info to an exception, don't catch the exception at lower levels of code. Instead catch it from the highest layer of code.